satpy/tests/reader_tests/test_satpy_cf_nc.py

```diff
@@ -0,0 +1,110 @@
+        vis006 = xr.DataArray(data_visir,
+                              dims=('y', 'x'),
+                              coords={'y': y_visir, 'x': x_visir, 'acq_time': ('y', time_vis006)},
+                              attrs={'name': 'image0', 'id_tag': 'ch_r06', 'coordinates': 'lat lon'})
+
+        ir_108 = xr.DataArray(data_visir,
+                              dims=('y', 'x'),
+                              coords={'y': y_visir, 'x': x_visir, 'acq_time': ('y', time_vis006)},
+                              attrs={'name': 'image1', 'id_tag': 'ch_tb11', 'coordinates': 'lat lon'})
+        qual_f = xr.DataArray(qual_data,
+                              dims=('y', 'z'),
+                              coords={'y': y_visir, 'z': z_visir, 'acq_time': ('y', time_vis006)},
+                              attrs={'name': 'qual_flags',
+                                     'id_tag': 'qual_flags'})
+        lat = xr.DataArray(lat,
+                           dims=('y', 'x'),
+                           coords={'y': y_visir, 'x': x_visir},
+                           attrs={'name': 'lat',
+                                  'standard_name': 'latitude'})
+        lon = xr.DataArray(lon,
+                           dims=('y', 'x'),
+                           coords={'y': y_visir, 'x': x_visir},
+                           attrs={'name': 'lon',
+                                  'standard_name': 'longitude'})
+        self.scene = Scene()
+        self.scene.attrs['sensor'] = ['avhrr-1', 'avhrr-2', 'avhrr-3']
+        scene_dict = {'image0': vis006,
+                      'image1': ir_108,
+                      'lat': lat,
+                      'lon': lon,
+                      'qual_flags': qual_f}
+        for key in scene_dict:
+            self.scene[key] = scene_dict[key]
+            self.scene[key].attrs.update(common_attrs)
+
+    def test_write_and_read(self):
+        """Save a file with cf_writer and read the data again."""
+        # '{testin}-{sensor}-{start_time:%Y%m%d%H%M%S}-{end_time:%Y%m%d%H%M%S}.nc'
+        filename = 'testingcfwriter{:s}-viirs-mband-20201007075915-20201007080744.nc'.format(
+            datetime.utcnow().strftime('%Y%j%H%M%S'))
+        self.scene.save_datasets(writer='cf',
+                                 filename=filename,
+                                 header_attrs={'instrument': 'avhrr'},
+                                 engine='h5netcdf',
+                                 flatten_attrs=True,
+                                 pretty=True)
+        scn_ = Scene(reader='satpy_cf_nc',
+                     filenames=[filename])
+        scn_.load(['image0', 'image1', 'lat'])
+        self.assertTrue(np.all(scn_['image0'].data == self.scene['image0'].data))
+        self.assertTrue(np.all(scn_['lat'].data == self.scene['lat'].data))  # lat loaded as dataset
+        self.assertTrue(np.all(scn_['image0'].coords['lon'] == self.scene['lon'].data))  # lon loded as coord
+        os.remove(filename)
```

Comment:
This is generating a Permission error on windows. I suspect the file isn't closed before removal, either because of the writting or the reading, or both... Could you try except the remove for now?